src/features/import/store.ts

```diff
@@ -1,16 +1,55 @@
-import { Column, ImportedFile } from './utils/types';
+import {
+  Column,
+  IMPORT_ERROR,
+  ImportedFile,
+  PersonImport,
+  PersonImportSummary,
+} from './utils/types';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 export interface ImportStoreSlice {
+  importResult: PersonImport;
+  importErrors: IMPORT_ERROR[];
+  preflightSummary: PersonImportSummary;
   pendingFile: ImportedFile;
 }
 
+const emptySummary: PersonImportSummary = {
+  addedToOrg: {
+    byOrg: {},
+    total: 0,
+  },
+  created: {
+    total: 0,
+  },
+  tagged: {
+    byTag: {},
+    total: 0,
+  },
+  updated: {
+    byField: {},
+    total: 0,
+  },
+};
+
 const initialState: ImportStoreSlice = {
+  importErrors: [],
+  importResult: {
```

Comment:
Shouldn't this be `null` until a report has been generated?

Reply:
good point!

Reply:
Cool! If this causes type problems in components (because `importResult` can now be `null`) consider hiding that fact away behind the relevant hook. If you, the programmer, know that it should never happen that the relevant components render while this is `null`, it's safe to throw an error for `null`. That allows you to return a non-null value (in an if statement checking for `null`).